ocaml/stdlib/camlinternalOO.ml

```diff
@@ -16,13 +16,18 @@
 
 open! Stdlib
 
-[@@@ocaml.flambda_o3]
-
 [@@@ocaml.inline 0]
 [@@@ocaml.afl_inst_ratio 0]
 
 open Obj
 
+(* Prevent confusion between arrays and blocks *)
+module Array = struct
+  include Array
```

Comment:
The 'include Array' seems dubious here. I think what you're aiming for is to provide functions called 'Array.unsafe_set' and 'Array.unsafe_get' that can safely be used on blocks, but the 'include Array' also brings in 'Array.get' and 'Array.set' (which, if I understand correctly, are unsafe in this context).

Rather than opaque_identity, should these functions be Obj.field? What type are they used at?

Reply:
These appear to be used at type `array`, but I'm not sure if the underlying value is actually an array -- there is so much `Obj.magic` around that it's hard to follow.